backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/predicate/impl/LuceneKnnPredicate.java

```diff
@@ -81,10 +85,24 @@ public void k(int k) {
 		}
 
 		@Override
-		@SuppressWarnings("unchecked")
 		public void vector(Object vector) {
+			this.vectorProvider = simple(
+					objectToVector( vector, vectorCodec, absoluteFieldPath, vectorElementsType, indexedVectorsDimension ) );
+		}
+
+		@Override
+		public void vectorParam(String parameterName) {
+			this.vectorProvider =
+					parameter( parameterName, Object.class, vector -> objectToVector( vector, vectorCodec, absoluteFieldPath,
+							vectorElementsType, indexedVectorsDimension ) );
```

Comment:
it seems safe to have such lambdas keeping scope/codec etc, since those are around for a long time, but let me know if that's not the case 😃